### ISSUE TITLE
[ENH] coordination discussion on foundation models, deep learning, backends

### ISSUE THREAD

**Issue by fkiraly** [sktime/sktime#6381]
Opening this issue to coordinating the various usmmer projects in relation toe foundation models, deep learning, backends, interfaces.

Below a list of related umbrella issues and individual issues - for now, focusing on forecasting primarily.

FYI @Abhay-Lejith, @fnhirwa, @geetu040, @julian-fong, @pranavvp16, @Xinyu-Wu-0000.

- umbrella issue - foundation model zoo - https://github.com/sktime/sktime/issues/6177
    - issue subtype one: interfacing individual models and frameworks, eg., hugging face, chronos, moirai
    - issue subtype one: API extension for foundation models: handling "global model" train/predict, pretrained
 - umbrella issue - data containers - https://github.com/sktime/sktime/issues/6091
    - from this, my opinion on highest priority: `polars` (due to `sklearn` and nixtla-verse adoption)
    - parallel/distributed functionality should also be added to broadcasting (prio), and compositors (e.g., pipeline)
- framework and "favourite model" interfaces:
    - pytorch-forecasting - https://github.com/sktime/sktime/issues/4923
    - darts - https://github.com/sktime/sktime/issues/1624 
    - autots - https://github.com/sktime/sktime/issues/5406
    - greykite - https://github.com/sktime/sktime/issues/1621
- deep learning backend API and architecture considerations
    - refactoring DL backend interface - priority on pytorch
    - model persistence and serialization - including nested serialization

**Comment by yarnabrina** [comment]
Another darts PR: https://github.com/sktime/sktime/pull/5043

**Comment by fkiraly** [comment]
If we go by the topics proposed in GSoC proposals and previous work, we get - for the start:

- @fnhirwa -> darts, pytorch, pytorch-forecasting
- @geetu040  -> pytorch, pytorch-forecasting; hugging face, peft/lora
- @pranavvp16 -> polars, sklearn/polars, nixtla/polars; hugging face, peft/lora
- @julian-fong -> pre-trained models, fine-tuning, fastAI like API
- @Xinyu-Wu-0000 -> global API, pre-training and fine-tuning examples

There are three intersections here:

* pytorch-forecasting focus was suggested by multiple people - of course this is important, so no surprise if multiple smart people come to the same conclusion: @fnhirwa, @geetu040, @pranavvp16 
* hugging face interface, @geetu040, @pranavvp16
* global forecating, pre-training API - @Xinyu-Wu-0000 and @julian-fong

One possible assignment that avoids conditionalities and duplications for the 1st month would be:

- @fnhirwa -> darts, pytorch-forecasting ("framework interfacing" topic, pytorch backend topic)
- @geetu040 -> hugging face, peft/lora, scattered foundation models ("foundation models & API" topic)
- @pranavvp16 -> polars, sklearn/polars, nixtla/polars ("polars & distributed" topic)
- @Xinyu-Wu-0000 -> leads on global API design, works with @benHeid, based on https://github.com/sktime/sktime/pull/6228
- @julian-fong -> one of: pre-training/fine-tuning (fastAI like), or contribute to polars upgrade

**Comment by benHeid** [comment]
Possible further work item could be integration of GluonTS.

**Comment by fkiraly** [comment]
Yes - gluonts was actually part of the very original forecaster feature wishlist, here: https://github.com/sktime/sktime/issues/220
It's funny to see how long that wishlist was and how almost everything on it is now available - most recently the bagging ensemble which I upgraded due to `tsbootstrap` (FYI @astrogilda). In a similar vein, the newer "auto-gluon" is a "one size fits all" approach similar to autots.

`gluonts` has its own data container, there are already some converters in `sktime` (possibly incomplete): https://github.com/sktime/sktime/issues/2860

**Comment by fkiraly** [comment]
from May 10 meeting:

* pytorch-forecasting is connected both to global API project of @Xinyu-Wu-0000 and @fnhirwa 
    * @Xinyu-Wu-0000 PR #6228 will already provide full framework integration -> duplication with @fnhirwa subtpoc pytorch-forecasting
    * so @fnhirwa focus should be on `darts` in the first two weeks
* @yarnabrina comment: `darts` converts everything to internal container - container item; existing PR that could/should be picked up #5043

**Comment by pranavvp16** [comment]
As discussed in todays mentoring meet with @benHeid, and as mentioned here I will start working on adding support to polars scitype for the first month. Commenting this here to co-ordinate with other mentees and mentors, Please feel free to reply on this if any other mentee is also interested or working on adding polars support.

**Comment by fkiraly** [comment]
@pranavvp16, @julian-fong, a high-level outline is in this issue here: https://github.com/sktime/sktime/issues/5423#issue-1943129169

There are two things one could work in parallel:

* `sktime` needs mtypes implemented, we could start with `Series`, then `Panel`. Since `polars` has no multi-index (is this correct?) I would suggest the same conventions as in `dask`, for `Panel`.
* in `skpro`, a `polars` container format is already implemented, although this is not full support as internally it just converts back/forth.

(and imo these are currently the only two fully parallel items)

The "battle plan" for support is - in both packages - first mtype support, then enable support in a few estimators, see if we can support eager and even lazy.

I think `skpro` estimators are simpler, so if in parallel to working on `sktime` mtypes, we try enabling native `polars` support for a number of estimators, we will learn of the challenges and solutions ahead of time, and at a lower cognitive cost.

What are your thoughts? Any preferences?

**Comment by fkiraly** [comment]
Related to `polars`, there is also this refactor of the `datatypes` module:

https://github.com/sktime/sktime/pull/6033

This would make it easier to add mtypes with soft dependencies, and someone could pick it up and review - or complete - it. I was also going to look at it soon.

**Comment by pranavvp16** [comment]
yess the plan looks good to me for now, but polars doesn't even support index as well as multindex. Also if I'm not wrong we have to get this refactor merged before we can start adding polars mtype support in sktime ??

**Comment by fnhirwa** [comment]
As per the last conversation with @fkiraly I'll pick up this [[ENH] darts adapter #5043](https://github.com/sktime/sktime/pull/5043) and liaise with @yarnabrina 

Commenting here to coordinate so It can't collide with any other ongoing task.

**Comment by fkiraly** [comment]
Re polars, to get back to coordinating tasks, current discussion sounds like:

* @pranavvp16 will work on type checkers and converters for `sktime`, that is issue #5423. Can you kindly comment there that I can assign you?
* @julian-fong will work on `skpro` functionality to extend `polars` coverage using the existing data type. I will open an issue there.

Does this make sense, and does this align with your preferences?

**Comment by fkiraly** [comment]
> As per the last conversation with @fkiraly I'll pick up this [[ENH] darts adapter #5043](https://github.com/sktime/sktime/pull/5043) and liaise with @yarnabrina

Excellent - the issue is #1624, could you kindly comment there so I can assign you?

**Comment by fkiraly** [comment]
@julian-fong, `polars` issue in `skpro`: https://github.com/sktime/skpro/issues/342
